Small Enchancements + VTune - Main Loop 1 of N

fmatthew5876 opened the pull request:
I pulled some of the standalone commits out of #2017 to here. 

This adds some small improvements and includes the VTune integration. I made an attempt to put vtune frame instrumentation around our current main loop. It shows the CPU burn effect that gets fixed in #2017 

This is tested, ready for review and merge.

Ghabry commented:
For all this main loop stuff pls don't rush it. Want some cooperation with emscripten asynchify here so we can finally get this in.
(Will update the buildscript 2020)

fmatthew5876 commented:
I fixed up the vtune cmake file. It also works on windows now if you specify `-DVTUNE_ROOT=<VTUNE INSTALL PATH`

Ghabry commented on resources/easyrpg-player.6.adoc:
```diff
@@ -50,6 +50,9 @@ for some additional binary patches.
   Enable frames per second counter.
 
+*--fps-render-window*::
+  Render the frames per second counter both full screen and windowed mode.
```
counter _in_ both

fmatthew5876 replied:
fixed

fmatthew5876 commented:
> 
> 
> Well, I'm currently not seeing the big picture of this instrumentation stuff. Looking forward to the next PRs to make this look nicer :P

The Framerate histogram in vtune isn't working for me in this version. But it does later after the main loop gets restructured.

Addressed review comments and rebased

carstene1ns commented on src/scene.h:
```diff
@@ -198,7 +198,7 @@ class Scene {
 
 	/** Called by the graphic system to request drawing of a background, usually a system color background */
-	virtual void DrawBackground();
+	virtual void DrawBackground(Bitmap& dst);
```
dox?

fmatthew5876 commented:
Rebased and fixed comments
